mask/eolearn/mask/cloud_mask.py

```diff
@@ -292,3 +297,592 @@ def get_s2_pixel_cloud_detector(threshold=0.4, average_over=4, dilation_size=2,
+            self.multi_features = multi_features
+        else:
+            self.multi_features = (None, None)
+
+        self.mask_feature = mask_feature
+
+        # Set thresholding and morph. ops. parameters and kernels
+        self.mono_threshold = mono_threshold
+        self.multi_threshold = multi_threshold
+
+        if average_over is not None and average_over > 0:
+            self.avg_kernel = disk(average_over) / np.sum(disk(average_over))
+        else:
+            self.avg_kernel = None
+
+        if dilation_size is not None and dilation_size > 0:
+            self.dil_kernel = disk(dilation_size).astype(np.uint8)
+        else:
+            self.dil_kernel = None
+
+    @staticmethod
+    def _get_max(data):
+        """Timewise max for masked arrays."""
+        return np.ma.max(data, axis=0).data
+
+    @staticmethod
+    def _get_min(data):
+        """Timewise min for masked arrays."""
+        return np.ma.min(data, axis=0).data
+
+    @staticmethod
+    def _get_mean(data):
+        """Timewise mean for masked arrays."""
+        return np.ma.mean(data, axis=0).data
+
+    @staticmethod
+    def _get_std(data):
+        """Timewise std for masked arrays."""
+        return np.ma.std(data, axis=0).data
+
+    def _parse_resolution_data(self, reference_shape, meta_info):
+        """ Compute the resampling factor for height and width of the input array
+
+        :param reference_shape: Tuple specifying height and width in pixels of high-resolution array
+        :type reference_shape: tuple of ints
+        :param meta_info: Meta-info dictionary of input eopatch. Defines OGC request and parameters used to create the
+                            eopatch
+        :return: Rescale factor for rows and columns
+        :rtype: tuple of floats
+        """
+        # Figure out resampling size
+        height, width = reference_shape
+
+        service_type = ServiceType(meta_info['service_type'])
+
+        if service_type == ServiceType.WMS:
+            if self.processing_resolution is not None:
+                pres_x, pres_y = self.processing_resolution
+                rescale = (pres_y / height, pres_x / width)
+            else:
+                rescale = (1., 1.)
+
+            # Use default sigma value, since resolution is unknown
+            sigma = 1.0
+
+        elif service_type == ServiceType.WCS:
+            hr_res_x, hr_res_y = float(meta_info['size_x'].strip('m')), float(meta_info['size_y'].strip('m'))
+
+            if self.processing_resolution is not None:
+                pres_x, pres_y = [float(res.strip('m')) for res in self.processing_resolution]
+                pres = (pres_x + pres_y) / 2.0
+
+                rescale = (hr_res_y / pres_y, hr_res_x / pres_x)
+            else:
+                pres = (hr_res_x + hr_res_y) / 2.
+                rescale = (1., 1.)
+
+            sigma = 100. / pres
+
+        return rescale, sigma
```

Comment:
1) If WMS and WCS are the only possible choices, then end with `else`, otherwise add an extra else for an unknown/incompatible service type.

2) In case that it's neither a WCS nor WMS, the `rescale` and `sigma` parameters are potentially referenced before assignment. Perhaps it's better to set some default values at the appropriate scope level first and set them later if the appropriate conditions are satisfactory.

3)
instead of
```python
if condition:
    var = something
else:
    var = something_generic
```

you can do 

```python
var = something_generic
if condition:
    var = something
```

it's the same, but more intuitive

Reply:
Added else for unknown service types. I used default variable values as you mentioned where I thought appropriate.